### PR TITLE
add program notifications

### PR DESCRIPTION
**Pull request by dringler**
add program notification extension for pycon/pydata

### CODE REVIEW

File: EuroPythonBot/extensions/programme_notifications/__init__.py [hunk 1]
```diff
@@ -15,6 +15,9 @@
 
 _logger = logging.getLogger(f"bot.{__name__}")
 
+# timestamp during the conference used for testing (see timewarm in config)
+TIME_DURING_CONFERENCE = "2024-04-23T15:57:45+02:00"
```
Comment (theopinard) [inline EuroPythonBot/extensions/programme_notifications/__init__.py]:
```suggestion
TIME_DURING_CONFERENCE = "2024-04-23T15:54:45+02:00"
```

**Review by theopinard — APPROVED** [review]
LGTM :)

File: EuroPythonBot/extensions/programme_notifications/__init__.py (hunk 1 above)
Comment (theopinard) [reply EuroPythonBot/extensions/programme_notifications/__init__.py]: if you look in the config you have a 5 min time before the notification but only 2 for the room. [code](https://github.com/PYCONDE/discord/blob/program-notifications-extension/EuroPythonBot/extensions/programme_notifications/services/notifier.py#L16-L17)
however in the fake time you use for the conference, you set it to 2 min and 15 seconds before the next talks [code](https://github.com/PYCONDE/discord/blob/program-notifications-extension/EuroPythonBot/extensions/programme_notifications/__init__.py#L19).

File: EuroPythonBot/config.toml [hunk 2]
```diff
@@ -29,83 +29,95 @@ LOG_LEVEL = "INFO"
...
+# B07-B08: 2580
+[programme_notifications.rooms.2580]
+webhook_id = "ROOM_2580"
+discord_channel_id = "1225454000800006144"
+
+[programme_notifications.rooms.2580.livestreams]
+# TODO: update URLs
+"2024-04-22" = "https://2024.pycon.de/live"
+"2024-04-23" = "https://2024.pycon.de/live"
+"2024-04-24" = "https://2024.pycon.de/live"
```
Comment (theopinard) [inline EuroPythonBot/config.toml:80]: btw the repo is open so anyone with a github account could get access to the live stream if they want to.

Comment (dringler) [reply EuroPythonBot/config.toml:80]: Good point. We should think about how to deal with the live stream links when we want to include them

**Pull request merged**